### PR TITLE
Multi ip

### PR DESCRIPTION
**Pull request by notedit**
Add multi ip support for endpoint.

It is useful for server have multi ip,  and  do server-to-server relay in  one nat.

### CODE REVIEW

**Comment by notedit** [comment]
@murillo128  Hi will we merge this?

**Comment by murillo128** [comment]
still out this week,  will check it next one

**Comment by murillo128** [comment]
LGTM could you add a few test cases on `test\endpoint.js` ?

File: lib/Endpoint.js [hunk 1]
```diff
@@ -32,17 +32,22 @@ class Endpoint
...
 		//Start it
 		this.bundle.Init();
 		//Store all transports
 		this.transports = new Set();
-		//Create candidate
-		this.candidate = new CandidateInfo("1", 1, "UDP", 33554431, ip, this.bundle.GetLocalPort(), "host");
+		//Create candidates 
+		this.candidates = [];
+		for (let i=0; i<this.ips.length; i++) {
+			let priority = Math.pow(2,24) * (126-i) + (65535-i)*Math.pow(2,8) + 255;
```
Comment (murillo128) [inline lib/Endpoint.js]: Shouldn't it be?
```
(2^24)*(type preference) +
              (2^8)*(local preference) +
              (2^0)*(256 - component ID)

type preference =126;
local preference = 65535 -i
component ID = 255;
```
so
```
let priority = Math.pow(2,24)*126 + Math.pow(2,8)*(65535-i) + 255;
```

Comment (notedit) [reply lib/Endpoint.js]: Yes.  I want the first ip's priority is bigger than the second, so i change the type preference.
In my testing,  priority is not the primary key,  the client side will  also consider network-cost  to do candidate pairing.

Comment (murillo128) [reply lib/Endpoint.js]: but all candidates are the same type, what changes is just the local preference.

Comment (notedit) [reply lib/Endpoint.js]: I will fix that.

**Comment by murillo128** [comment]
made cosmetic changes, fixed the priority and added tests here:  https://github.com/medooze/media-server-node/commit/1fcb3999a8ee6c0e3592dd8291fcb6326d462661

Thanxs for the contribution!

**Comment by murillo128** [comment]
Hi @notedit!  we lost the CLAs signed in clahub, would you mind singing the following CLA for your contributions to the Medooze projects?

https://cla-assistant.io/medooze/clas

**Pull request merged**